webapp/apps/taxbrain/helpers.py

```diff
@@ -30,18 +31,45 @@
 DEC_POINT = pp.Word('.', exact=1)
 FLOAT_LIT_FULL = pp.Word(pp.nums + '.' + pp.nums)
 COMMON = pp.Word(",", exact=1)
-REVERSE = pp.Word("<") + COMMON
-
+REVERSE = pp.Word("<")
 VALUE = WILDCARD | NEG_DASH | FLOAT_LIT_FULL | FLOAT_LIT | INT_LIT
-MORE_VALUES = COMMON + VALUE
-
 BOOL = WILDCARD | TRUE | FALSE
-MORE_BOOLS = COMMON + BOOL
-INPUT = (pp.Optional(REVERSE) +
-         BOOL +
-         pp.ZeroOrMore(MORE_BOOLS) | pp.Optional(REVERSE) +
-         VALUE +
-         pp.ZeroOrMore(MORE_VALUES))
+
+
+def is_safe(s):
+    """
+    Test if a string of comma-separated-values is "safe"
+    - is the string less than 100 characters?
+    - is the string boolean?
+    - is the string an integer or float?
+    - is the string a wildcard (*)?
+    - is the string a reverse character (<)?
+    If one of the tokens does not answer one of the above questions in the
+    affirmative, then it is deemed "not safe"
+
+    Returns:
+        success: whether value is "safe" or not
+    """
+    assert len(s) < 100
```

Comment:
I think it would be better to raise some other kind of exception here and only use `assert` for internal invariants.

Reply:
Ah, good catch. I changed this to return `False` instead. That's what I meant to do in the first place.